src/main/java/seedu/duke/Parser.java

```diff
@@ -0,0 +1,97 @@
+package seedu.duke;
+
+import seedu.duke.command.AddCommand;
+import seedu.duke.command.Command;
+import seedu.duke.command.DeleteCommand;
+import seedu.duke.command.DoneCommand;
+import seedu.duke.command.ExitCommand;
+import seedu.duke.command.FindCommand;
+import seedu.duke.command.ListCommand;
+import seedu.duke.exception.DukeEmptyDescriptionException;
+import seedu.duke.exception.DukeException;
+import seedu.duke.exception.DukeNoKeywordException;
+import seedu.duke.exception.DukeWrongCommandException;
+
+import java.time.LocalDate;
+import java.util.regex.Pattern;
+
```

Comment:
Could possibly include Javadoc for this class and its methods considering there are lots of details.